### PR TITLE
doc: Cleanup and enhance C lib and tracing documentation

### PR DESCRIPTION
**Pull request by nashif**
minor cleanup and restructuring for documentation of tracing and c library.

### CODE REVIEW

File: doc/guides/tracing/index.rst [hunk 1]
```diff
@@ -23,17 +26,258 @@ any application for platforms that have the required hardware support.
...
+  }
+
+How to serialize and emit fields as well as handling alignment, can be done
+internally and statically at compile-time in the bottom-layer.
+
+
+How to Activate?
+================
+
+Make sure ``CONFIG_TRACING_CTF=y`` is set (``CONFIG_TRACING_CTF_BOTTOM_POSIX=y``
```
Comment (dbkinder) [inline doc/guides/tracing/index.rst]: Let's link to the config option documentation:
```
Make sure :option:`CONFIG_TRACING_CTF` ``= y``
is set (:option:`CONFIG_TRACING_CTF_BOTTOM_POSIX`
``= y`` is selected by default when using
:option`CONFIG_BOARD_NATIVE_POSIX`).
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
Didn't like my intro sentence suggestion, eh?  OK.

File: doc/guides/tracing/index.rst (hunk 1 above)
Comment (nashif) [reply doc/guides/tracing/index.rst]: > Didn't like my intro sentence suggestion, eh? OK.

what intro sentence?

**Comment by nashif** [comment]
> Didn't like my intro sentence suggestion, eh? OK.

what intro sentence?

**Comment by dbkinder** [comment]
Changing:
```
The minimal C library is part of Zephyr and provides a minimal set of C
functions needed by Zephyr.

The following functions are implemented in the minimal C
library included with Zephyr:
```

to just:
```
The following functions are implemented in the minimal C
library included with Zephyr:
```

**Comment by nashif** [comment]
> Changing:

I think the first paragraph is essential. Your change request did not indicate you wanted all replaced. anyways..

**Pull request merged**